User-Friendly Variable Names for XFEM GeometricCutUserObjects

jbtompkins opened the pull request:
Geometric variables more easily recognizable to users.
 - Changed input variable syntax to more obvious naming scheme for XFEM GeometricCutUserObjects
 - Updated tests to use new syntax

closes #9603

### Complete each of the following items before creating a PR

- Include any relevant design information for your change
- Follow [Coding Standards](http://mooseframework.org/wiki/CodeStandards/)
- Submit one or more [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/) or modify existing test cases
- Reference or close a specific issue (refs # or closes #)

moosebuild commented:
Job [Precheck](https://www.moosebuild.org/job/106754/) on 827c1ae wanted to post the following:

Your code requires style changes.
In the top level of your repository, you can run:

`git clang-format 64a65906bd0b6c0e92f3211aef929dec189d1680`

Alternatively, apply the patch found [here](http://mooseframework.org/docs/PRs/9635/style.patch).
You can directly apply the patch by running:

`curl -s http://mooseframework.org/docs/PRs/9635/style.patch | git apply -v`

moosebuild commented:
Job [Precheck](https://www.moosebuild.org/job/106763/) on b4f12ff wanted to post the following:

Your code requires style changes.
In the top level of your repository, you can run:

`git clang-format 64a65906bd0b6c0e92f3211aef929dec189d1680`

Alternatively, apply the patch found [here](http://mooseframework.org/docs/PRs/9635/style.patch).
You can directly apply the patch by running:

`curl -s http://mooseframework.org/docs/PRs/9635/style.patch | git apply -v`

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/106811/) on 9a01b47 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/9635/site)

permcody commented on modules/xfem/src/userobjects/RectangleCutUserObject.C:
```diff
@@ -21,30 +21,54 @@ validParams<RectangleCutUserObject>()
   // Add required parameters
-  params.addRequiredParam<std::vector<Real>>("cut_data",
-                                             "Vector of Real values providing cut information");
+  params.addRequiredParam<std::vector<Real>>(
```
Did you know that you can create an input parameter type `Point`? That's much more appropriate here since you are expecting X, Y, Z. Not X, Y, Z, W, Q, A....

permcody commented on modules/xfem/include/userobjects/RectangleCutUserObject.h:
```diff
@@ -26,7 +26,10 @@ class RectangleCutUserObject : public GeometricCut3DUserObject
+  std::vector<Real> _vertex_point1;
+  std::vector<Real> _vertex_point2;
+  std::vector<Real> _vertex_point3;
+  std::vector<Real> _vertex_point4;
```
Switch these all to `Point`. std::vector<Real> is the wrong type.

nuclear-wizard commented:
Getting some strange behavior from changing the input type to point. At David's request I turned on Valgrind testing in civet. Please wait until these issues are resolved before merging.

james-tompkins committed:
User-Friendly Variable Names for XFEM GeometricCutUserObjects

Geometric variables more easily recognizable to users.

 - Changed input variable syntax to more obvious naming scheme for XFEM GeometricCutUserObjects
 - Updated tests to use new syntax

 closes #9603

permcody commented:
@nuclear-wizard - Your last comment is a little unclear. Are you ready for a review?

andrsd commented:
@permcody He changed the types according to your suggestion, but got really weird behavior on his machine that looked like HW issues. Since this new version worked fine on the build boxes, I'd say this is ready for the review.

permcody commented on modules/xfem/include/userobjects/LineSegmentCutUserObject.h:
```diff
@@ -26,7 +26,8 @@ class LineSegmentCutUserObject : public GeometricCut2DUserObject
 protected:
-  std::vector<Real> _cut_data;
+  std::vector<Real> _start_point;
+  std::vector<Real> _end_point;
```
Switch to `Point`

permcody commented on modules/xfem/src/userobjects/EllipseCutUserObject.C:
```diff
@@ -21,30 +21,28 @@ validParams<EllipseCutUserObject>()
-  _vertices.push_back(Point(_cut_data[6], _cut_data[7], _cut_data[8]));
-
-  std::pair<Point, Point> rays = std::make_pair(_vertices[0] - _center, _vertices[1] - _center);
+  std::pair<Point, Point> rays =
```
"rays" really should be typed as `RealVectorValue`s not `Points`.

permcody commented on modules/xfem/src/userobjects/EllipseCutUserObject.C:
```diff
@@ -83,7 +79,8 @@ EllipseCutUserObject::isInsideCutPlane(Point p) const
   {
-    std::pair<Real, Real> xy_loc = std::make_pair(ray * _unit_vec1, ray * _unit_vec2);
+    std::pair<Real, Real> xy_loc =
+        std::make_pair(ray * _long_axis_unit_vector, ray * _short_axis_unit_vector);
```
`Point`

permcody commented on modules/xfem/src/userobjects/LineSegmentCutUserObject.C:
```diff
@@ -27,14 +31,18 @@ validParams<LineSegmentCutUserObject>()
-  : GeometricCut2DUserObject(parameters), _cut_data(getParam<std::vector<Real>>("cut_data"))
+  : GeometricCut2DUserObject(parameters),
+    _start_point(getParam<std::vector<Real>>("start_point")),
+    _end_point(getParam<std::vector<Real>>("end_point"))
```
`Point`

permcody commented on modules/xfem/src/userobjects/RectangleCutUserObject.C:
```diff
@@ -21,36 +21,38 @@ validParams<RectangleCutUserObject>()
-  for (unsigned int i = 0; i < num_vertices; ++i)
+  for (unsigned int i = 0; i < _vertices.size(); ++i)
     _center += _vertices[i];
   _center *= 0.25;
```
Magic number? I guess we are assuming rectangle here. It would be nice if this was a constant.

andrsd replied:
how about `_center *= 1. / _vertices.size()`?

permcody reviewed (changes requested):
Looks like you fixed most of the types, but I see a few more. There are many types and utilities in libMesh for doing just about every geometric operation you want.

nuclear-wizard commented:
Just updating the branch to rebase from devel. Not yet ready for review.

permcody commented:
Closing per the 30-day limit.
